emcc.py

```diff
@@ -1441,10 +1441,13 @@ def default_setting(name, new_default):
 
   # Wasm SjLj cannot be used with Emscripten EH
   if settings.SUPPORT_LONGJMP == 'wasm':
+    # DISABLE_EXCEPTION_THROWING is 0 by default for Emscripten EH throwing, but
+    # Wasm SjLj cannot be used with Emscripten EH. So we disable it here.
+    DISABLE_EXCEPTION_THROWING = 1;
```

Comment:
I think it would be good to still error if the user provided this flag. That is, changing the default value makes sense as this PR does, but we can still error on incompatible user-provided flags instead of silently modifying one of them, that seems safer I think.

(we have `settings_changes` to see what things the user asked to change)

Reply:
Is there anywhere we error out based on `settings_changes`? I searched emcc.py but couldn't find any. Also it is a list and not a map, so I'm wondering if this structure is meant to be used this way?

Reply:
Oh, sorry, I remembered wrong. The actual thing is `settings_map` which is computed from that. Example usage:
```python
  if 'DISABLE_EXCEPTION_CATCHING' in settings_map and 'EXCEPTION_CATCHING_ALLOWED' in settings_map:
    # If we get here then the user specified both DISABLE_EXCEPTION_CATCHING and EXCEPTION_CATCHING_ALLOWED
    # on the command line.  This is no longer valid so report either an error or a warning (for
    # backwards compat with the old `DISABLE_EXCEPTION_CATCHING=2`
    if settings_map['DISABLE_EXCEPTION_CATCHING'] in ('0', '2'):
      diagnostics.warning('deprecated', 'DISABLE_EXCEPTION_CATCHING=X is no longer needed when specifying EXCEPTION_CATCHING_ALLOWED')
    else:
      exit_with_error('DISABLE_EXCEPTION_CATCHING and EXCEPTION_CATCHING_ALLOWED are mutually exclusive')
```